pyccel/codegen/compiling/compilers.py

```diff
@@ -298,14 +298,15 @@ def compile_program(self, compile_obj, output_folder, verbose = False):
         # Get compile options
         exec_cmd, includes, libs_flags, libdirs_flags, m_code = \
                 self._get_compile_components(compile_obj, accelerators)
+        linker_libdirs_flags = ['-Wl,-rpath' if l == '-L' else l for l in libdirs_flags]
 
         if self._info['language'] == 'fortran':
             j_code = (self._info['module_output_flag'], output_folder)
         else:
             j_code = ()
 
         cmd = [exec_cmd, *flags, *includes, *libdirs_flags,
-                *m_code, compile_obj.source,
+                 *linker_libdirs_flags, *m_code, compile_obj.source,
                 '-o', compile_obj.program_target,
                 *libs_flags, *j_code]
```

Comment:
Are we not duplicating most of the flags contained in `libdirs_flags`?

Reply:
We should be duplicating all of them. The idea is to do: `-L libdir -Wl,-rpath libdir`. The first command `-L libdir` tells the compiler where to look for libraries. The second command `-Wl,-rpath libdir` tells the linker to save this path in the executable. Without the first command the file will not compile and will raise errors like:
`undefined reference to omp_get_thread_num_ `
Without the second command, the file will compile, but it will not know where to look for the libraries (only a problem when linking shared libraries). It will therefore raise errors like:
`cannot open shared object file: No such file or directory`
on execution

Reply:
I see, thanks!